src/github.rs

```diff
@@ -1863,12 +1900,14 @@ impl GithubClient {
         let req = req
             .build()
             .with_context(|| format!("failed to build request {:?}", req_dbg))?;
+        let test_capture_info = test_record::capture_request(&req);
```

Comment:
I wonder if it would be better to have a wrapper around `self.client`? Seems more composable.

Reply:
I'm not sure I understand the suggestion. `Client` has a pretty big surface area, and I'm not sure what would be wrapped or why. Can you say more?

Reply:
Actually, I guess this is right :)